frontend/src/metabase/parameters/components/FormattedParameterValue/FormattedParameterValue.tsx

```diff
@@ -20,22 +26,59 @@ function FormattedParameterValue({
   value,
   placeholder,
 }: FormattedParameterValueProps) {
+  const { data, isLoading } = useGetParameterValuesQuery(
+    {
+      parameter: normalizeParameter(parameter),
+      field_ids: getNonVirtualFields(parameter).map(field => Number(field.id)),
+    },
+    {
+      skip:
+        !parameter ||
+        parameterHasNoDisplayValue(value) ||
+        Boolean(parameter.values_source_config?.values) ||
+        (Array.isArray(value) && value.length !== 1),
```

Comment:
This condition is unclear, can we extract this part and give it a meaningful name?